### PR TITLE
Improve how Hinge/Prismatic joints handle the connection to JointRequest bus

### PR DESCRIPTION
**Pull request by moraaar**
Signed-off-by: moraaar <moraaar@amazon.com>

## What does this PR do?

Before this change the Hinge & Prismatic joints were connecting to the JointRequest bus too early, when their native joint wasn't created yet, so potentially JointRequest calls could arrive before the native joint is created.

## How was this PR tested?

PhysX unit tests passed.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

File: Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp [hunk 1]
```diff
@@ -21,8 +21,7 @@ namespace PhysX
             serializeContext->Class<EditorStaticRigidBodyComponent, AzToolsFramework::Components::EditorComponentBase>()
                 ->Version(1);
 
-            AZ::EditContext* editContext = serializeContext->GetEditContext();
-            if (editContext)
+            if (auto* editContext = serializeContext->GetEditContext())
```
Comment (hultonha) [inline Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp:24]: uber nit: (could do the same above for the `serializeContext` if you haven't started AR yet)

**Review by hultonha — APPROVED** [review]
Great updates! 😄

File: Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp (hunk 1 above)
Comment (moraaar) [reply Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp:23]: It goes to a staging branch, there is no AR involved yet. Change done.

Comment (hultonha) [reply Gems/PhysX/Code/Source/EditorStaticRigidBodyComponent.cpp:23]: Awesome thanks!

**Pull request merged**